pepper-apis/src/main/java/org/broadinstitute/ddp/route/Auth0LogEventRoute.java

```diff
@@ -0,0 +1,139 @@
+import static org.broadinstitute.ddp.constants.RouteConstants.Header.AUTHORIZATION;
+import static org.broadinstitute.ddp.constants.RouteConstants.Header.BEARER;
+import static org.slf4j.LoggerFactory.getLogger;
+
+
+import com.typesafe.config.Config;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.json.auth0.Auth0LogEvent;
+import org.broadinstitute.ddp.json.errors.ApiError;
+import org.broadinstitute.ddp.service.Auth0LogEventService;
+import org.broadinstitute.ddp.util.ResponseUtil;
+import org.broadinstitute.ddp.util.SystemPropertyUtil;
+import org.slf4j.Logger;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+/**
+ * Handler for POST ../auth0-log-event?tenant=[TENANT_NAME]
+ *
+ * <p>JSON (sent in payload) is parsed and auth0 log events are extracted from it.
+ * Then log events are logged and persisted into table 'auth0_log_event'.
+ *
+ * <p>Header ("Authorization") with authorization token can be (optionally) specified in
+ * Auth0 Custom Webhook definition.<br>
+ * If config parameter 'privateApisTokens.auth0LogEventBearerToken' is specified and not empty
+ * then it contains a token which should be compared with token in header 'Authorization'
+ * (in a format: 'Bearer ' + 'privateApisTokens.auth0LogEventBearerToken').
+ * Token check can be disabled with setting system property 'auth0LogEvent.tokenCheck.disable'=true;
+ * If token check is enabled and specified in config then the same token should be specified in
+ * Auth0 Custom Webhook settings (with prefix 'Bearer ').
+ */
+public class Auth0LogEventRoute implements Route {
+
+    private static final Logger LOG = getLogger(Auth0LogEventRoute.class);
+
+    /**
+     * It is possible to disable auth0 log event URL token check via system property
+     * (by default it is enabled and defined by config params).<br>
+     * If system property 'auth0LogEvent.tokenCheck.disable' = true then config params are ignored.
+     */
+    public static final String SYSTEM_PROPERTY_AUTH0_LOG_EVENT_TOKEN_CHECK_DISABLE = "auth0LogEvent.tokenCheck.disable";
+    /** By default token check is not disabled */
+    public static final boolean DEFAULT_AUTH0_LOG_EVENT_TOKEN_CHECK_DISABLE = false;
+
+    /** Mandatory parameter in the log event URL. Specifies name of auth0 tenant */
+    public static final String QUERY_PARAM_TENANT = "tenant";
+
+    private final Config config;
+
+    public Auth0LogEventRoute(final Config config) {
+        this.config = config;
+    }
+
+    @Override
+    public Object handle(Request request, Response response) throws Exception {
+        String tenant = readTenant(request);
+        checkAuthorizationToken(request);
+        checkBody(request);
+
+        final var auth0LogEventService = new Auth0LogEventService();
+
+        final var logEvents = auth0LogEventService.parseAuth0LogEvents(request.body());
```

Comment:
Coding standards question: usually we only declare fields `final` (or variables that need to be final for the sake of lambdas).  Should we formalize the non-use of `final`s here?

Reply:
Yes, the finals not necessary here (should I better remove it?), I just trying to follow some of existing code practices: assigning final prevents from occasional changing the variable and make it more safe in a concurrent env. I usually set non-final when I really know that a variable could be changed..
Extra `final` keyword makes code a bit more verbose: unfortunately they didn't added in Java11 `val` (which exist in Scala and Kotlin denoting an 'immutable variable').

Reply:
Removed `final` keyword